pkg/pillar/cmd/zedrouter/server.go

```diff
@@ -336,3 +345,41 @@ func (hdl openstackHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	w.WriteHeader(http.StatusNotFound)
 }
+
+// ServeHTTP for kubeConfigHandler provides cluster kube config
+func (hdl kubeConfigHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		msg := "kube config handler: request method is not Post"
+		log.Error(msg)
+		http.Error(w, msg, http.StatusMethodNotAllowed)
+		return
+	}
+	if r.Header.Get("Content-Type") != "application/json" {
+		msg := "kube config handler: Content-Type header is not application/json"
+		log.Error(msg)
+		http.Error(w, msg, http.StatusUnsupportedMediaType)
+		return
+	}
+	kubeConfig, err := ioutil.ReadAll(r.Body)
```

Comment:
This is missing the size limit check. You should be able to use the pattern from utils which is
	content := make([]byte, maxReadSize)
	n, err := r.Read(content)
	if err != nil {
		err = fmt.Errorf("StatAndRead %s failed: %v", filename, err)
		if log != nil {
			log.Error(err)
		}
		return "", fi.ModTime(), err
	}
	if n == maxReadSize {
		err = fmt.Errorf("StatAndRead %s truncated after %d bytes",
			filename, maxReadSize)